kernel/tests/write.rs

```diff
@@ -0,0 +1,134 @@
+use std::sync::Arc;
+
+use arrow::array::StringArray;
+use arrow::record_batch::RecordBatch;
+use arrow_schema::Schema as ArrowSchema;
+use arrow_schema::{DataType as ArrowDataType, Field};
+use object_store::memory::InMemory;
+use object_store::path::Path;
+use object_store::ObjectStore;
+use serde_json::{json, to_vec};
+use url::Url;
+
+use delta_kernel::engine::arrow_data::ArrowEngineData;
+use delta_kernel::engine::default::executor::tokio::TokioBackgroundExecutor;
+use delta_kernel::engine::default::DefaultEngine;
+use delta_kernel::schema::{DataType, SchemaRef, StructField, StructType};
+use delta_kernel::Table;
+
+// setup default engine with in-memory object store.
+fn setup(
+    table_name: &str,
+) -> (
+    Arc<dyn ObjectStore>,
+    DefaultEngine<TokioBackgroundExecutor>,
+    Url,
+) {
+    let table_root_path = Path::from(format!("/{table_name}"));
+    let url = Url::parse(&format!("memory:///{}/", table_root_path)).unwrap();
+    let storage = Arc::new(InMemory::new());
+    (
+        storage.clone(),
+        DefaultEngine::new(
+            storage,
+            table_root_path,
+            Arc::new(TokioBackgroundExecutor::new()),
+        ),
+        url,
+    )
+}
+
+// we provide this table creation function since we only do appends to existing tables for now.
+// this will just create an empty table with the given schema. (just protocol + metadata actions)
+async fn create_table(
+    store: Arc<dyn ObjectStore>,
+    table_path: Url,
+    schema: SchemaRef,
+    partition_columns: &[&str],
+) -> Result<Table, Box<dyn std::error::Error>> {
+    let table_id = "test_id";
+    let schema = serde_json::to_string(&schema)?;
+
+    let protocol = json!({
+        "protocol": {
+            "minReaderVersion": 3,
+            "minWriterVersion": 7,
+            "readerFeatures": [],
+            "writerFeatures": []
+        }
+    });
+    let metadata = json!({
+        "metaData": {
+            "id": table_id,
+            "format": {
+                "provider": "parquet",
+                "options": {}
+            },
+            "schemaString": schema,
+            "partitionColumns": partition_columns,
+            "configuration": {},
+            "createdTime": 1677811175819u64
+        }
+    });
+
+    let data = [
+        to_vec(&protocol).unwrap(),
+        b"\n".to_vec(),
+        to_vec(&metadata).unwrap(),
+    ]
+    .concat();
+
+    // put 0.json with protocol + metadata
+    let path = table_path.path();
+    let path = format!("{path}_delta_log/00000000000000000000.json");
+    store.put(&Path::from(path), data.into()).await?;
+    Ok(Table::new(table_path))
+}
+
+#[tokio::test]
```

Comment:
I think you can setup tracing using `#[test_log::test(tokio::test)]` and avoid the try_init. [Relevant docs for test_log](https://docs.rs/test-log/latest/test_log/attr.test.html)

Reply:
hm I'm curious on peoples' thoughts on this.. seems like the init is a simple step but maybe it is nice to use test_log? I haven't been using it since it feels clearer for me to use typical `#[tokio::test]` and just to manual init

Reply:
Yeah I think I could go either way on this. This macro isn't really saving us from too much code. I think what I would like is a consistent approach in the codebase. Currently we use `test_log` in `snapshot`, `scan/mod`, and `arrow_expression.rs`.

Reply:
punting https://github.com/delta-incubator/delta-kernel-rs/issues/417